Shell command module logs lines as format strings

Seen in Kuma tests, which use terratest with gingo:

```
Test Multizone Resilience for K8s should see global entities in zone after a global restart 2021-09-03T09:09:21+10:00 retry.go:91: install yaml resource
Test Multizone Resilience for K8s should see global entities in zone after a global restart 2021-09-03T09:09:21+10:00 logger.go:66: Running command kubectl with args [--kubeconfig /home/jpeach/.kube/kind-kuma-1-config apply -f /tmp/Test%20Multizone%20Resilience%20for%20K8s%20should%20see%20global%20entities%20in%20zone%20after%20a%20global%20restart167549003]
Test Multizone Resilience for K8s should see global entities in zone after a global restart 2021-09-03T09:09:21+10:00 logger.go:66: Error from server (InternalError): error when creating "/tmp/Test%!M(MISSING)ultizone%!R(MISSING)esilience%!f(MISSING)or%!K(MISSING)8s%!s(MISSING)hould%!s(MISSING)ee%!g(MISSING)lobal%!e(MISSING)ntities%!i(MISSING)n%!z(MISSING)one%!a(MISSING)fter%!a(MISSING)%!g(MISSING)lobal%!r(MISSING)estart167549003": Internal error occurred: failed calling webhook "mesh.defaulter.kuma-admission.kuma.io": Post "https://kuma-control-plane.kuma-system.svc:443/default-kuma-io-v1alpha1-mesh?timeout=10s": dial tcp 10.96.184.11:443: connect: connection refused
Test Multizone Resilience for K8s should see global entities in zone after a global restart 2021-09-03T09:09:21+10:00 retry.go:103: install yaml resource returned an error: error while running command: exit status 1; Error from server (InternalError): error when creating "/tmp/Test%20Multizone%20Resilience%20for%20K8s%20should%20see%20global%20entities%20in%20zone%20after%20a%20global%20restart167549003": Internal error occurred: failed calling webhook "mesh.defaulter.kuma-admission.kuma.io": Post "https://kuma-control-plane.kuma-system.svc:443/default-kuma-io-v1alpha1-mesh?timeout=10s": dial tcp 10.96.184.11:443: connect: connection refused. Sleeping for 3s and will try again.
```
Note the 3rd log line, where the format string interpolation fails.

The problem is here:

https://github.com/gruntwork-io/terratest/blob/eca92144b6fc75ff55d42214a9b6ddef9345914c/modules/shell/command.go#L179

--- a/modules/shell/command.go
+++ b/modules/shell/command.go
@@ -176,7 +176,7 @@ func readData(t testing.TestingT, log *logger.Logger, reader *bufio.Reader, writ
 			break
 		}
 
-		log.Logf(t, line)
+		log.Logf(t, "%s", line)
 		if _, err := writer.WriteString(line); err != nil {
 			return err
 		}

--- a/modules/shell/command.go
+++ b/modules/shell/command.go
@@ -176,7 +176,14 @@ func readData(t testing.TestingT, log *logger.Logger, reader *bufio.Reader, writ
 			break
 		}
 
+		// logger.Logger has a Logf method, but not a Log method.
+		// We have to use the format string indirection to avoid
+		// interpreting any possible formatting characters in
+		// the line.
+		//
+		// See https://github.com/gruntwork-io/terratest/issues/982.
 		log.Logf(t, "%s", line)
+
 		if _, err := writer.WriteString(line); err != nil {
 			return err
 		}

